### DISCUSSION TITLE
NUTS translation in Spanish

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#672] [Q&A] [answered]
We are generating our eForms in Spanish and when we render the physical model for visualization the nuts do not appear correctly.

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/9e61ec11-dd14-4082-8e8d-4212ba74400d)

They do not appear correctly in the eNotices2 application or when rendering through the swagger render operation (https://viewer.preview.ted.europa.eu/swagger-ui/index.html#/notice-render-controller/renderNotice) .

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/9ac649b0-0a0d-4e98-8fc9-d89145c2f0c1)

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/35227121-4a78-465b-a3f6-295f03884e18)

On the other hand, if in the render operation we indicate that it be displayed in English, it appears fine.

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/68de44b4-c63d-453a-908f-bd062c4a50a9)

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/e7e6385b-d0f0-4a74-8050-6ff461b1d9ef)

Is there something wrong with the transformation process for visualization? Or are we not generating the physical model well?

![image](https://github.com/OP-TED/eForms-SDK/assets/115476802/00ed8ea3-c141-4181-8fce-033997f08e7f)

Thanks in advance

**Comment by bertrand-lorentz** [comment, marked as answer]
Thanks for your feedback.

The label for NUTS codes are not really translated, they are just the name of the region in the local language. Up until now, those labels were only present in the English translations (file code_en.xml).
In order to avoid the problem you describe, we will put those texts in all languages. See commit at https://github.com/OP-TED/eForms-SDK/commit/5aedfab419ed8b747f82caa2446240037487ed38

This will be available in SDK 1.9.0, and then included in a patch version for previous supported SDK versions.

**Reply by AlexAS81** [reply]
And if we send eforms in Spanish in production, will it look correct or until that patch is included in the SDKs prior to 1.9, will it continue to look bad?

**Reply by AlexAS81** [reply]
How will the production environment be in a month to be able to correctly display this type of information with eForms generated with SDK1.7?

**Reply by ferrakaop** [reply]
The applications (yours and ours) just have to find the (language-independent) NUTS labels from the code_en.xml file - that how it works in production for our viewer and will continue to work for now.  Adding (repeating) the NUTS code labels in code_xx.xml for all languages will just make it simpler in future as there will no need to know that NUTS labels are only in the code_en.xml file.

**Reply by AlexAS81** [reply]
I still have a doubt. 
When we send our eForm (.xml file) will it be seen correctly with its NUTS translated into Spanish in the search engine https://ted.europa.eu/TED/search/search.do? That is, when we access the publication of the notice through the publication URL, will it be seen correctly in Spanish?

**Reply by ferrakaop** [reply]
Yes, it will be ok. Your XML only has the NUTS code as a code, then our viewer application (the same one is currently used on TED too) figures out how to show the label for the NUTS code (by looking it up in the code_en.xml file for now).
See for example, https://ted.europa.eu/udl?uri=TED:NOTICE:511185-2023:TEXT:ES:HTML (a Spanish eForms notice): the XML has <cbc:CountrySubentityCode listName="nuts">ES511</cbc:CountrySubentityCode> and the HTML shows it as "Girona"
